src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -260,7 +260,16 @@ public OptionSet Options
 
         internal void UpdateCurrentSolutionOnOptionsChanged()
         {
-            SetCurrentSolution(CurrentSolution.WithOptions(new SolutionOptionSet(_legacyOptions)));
+            // TODO: this should be locked as it is updating CurrentSolution.  However, that triggers a deadlock in options due to:
+            // https://github.com/dotnet/roslyn/issues/64681
+            //
+            // This should be fixable once SolutionCrawler is entirely removed.
+            // using (_serializationLock.DisposableWait())
```

Comment:
we can't call the overload either as it takes serializationLock, which will lead to the same deadlock as here :'(

Reply:
I didn't read that bug fully earlier, oh goodness that looks awful.

Reply:
But what prevents that from racing against workspace changes? Because if there's a different change between the read/write, aren't we just blowing that away? Or is this only going to work because that's early enough and nothing actually exists in this solution yet?

I'm not convinced this isn't Really Bad™ and we need an answer before soluton crawler goes away.

Reply:
this i "really bad" :)  not sure what an acceptable middle ground is.  we can't take the lock currently.  so our options are either:

1. do not take the lock.  have the potential for misordered events.
2. do not take the lock.  do not issue the event.  have the potential for a lost event.

We're currently in '2', and i think maybe we should keep it htat way for the time being since it's where we are currently.